app/uk/gov/hmrc/selfservicetimetopay/controllers/CalculatorController.scala

```diff
@@ -80,12 +80,16 @@ class CalculatorController(calculatorConnector: CalculatorConnector) extends Tim
   }
 
   def submitAmountsDue: Action[AnyContent] = Action.async { implicit request =>
+    var loggedIn = false
```

Comment:
Cool. Get what you're saying now. Simple case of needing to assemble the data you need *out* of the first map for use in the second:

```
    sessionCache.get.map {
      case Some(TTPSubmission(_, _, _, taxpayer, _, _, CalculatorInput(debits, _, _, _, _, _))) => (Option(debits), taxpayer.isDefined)
      case _ => (Option(IndexedSeq.empty), false)
    }.map {
      case (None, isLoggedIn) => BadRequest(amounts_due_form(CalculatorAmountsDue(IndexedSeq.empty),
        CalculatorForm.amountDueForm.withGlobalError("You need to add at least one amount due"), isLoggedIn))
      case _ => Redirect(routes.CalculatorController.getPaymentToday())
    }
```